src/main/kotlin/org/rust/ide/refactoring/RsRenameProcessor.kt

```diff
@@ -44,4 +46,42 @@ class RsRenameProcessor : RenamePsiElementProcessor() {
     }
 
     private fun String.ensureQuote(): String = if (startsWith('\'')) this else "'$this"
+
+    override fun createUsageInfo(element: PsiElement, ref: PsiReference, referenceElement: PsiElement): UsageInfo {
+        if (ref !is RsStructLiteralFieldReferenceImpl) {
+            return super.createUsageInfo(element, ref, referenceElement)
+        } else {
+            // handle renaming struct literal shorthand identifier
+            val renameSourceType = when (element) {
+                is RsPatBinding -> RenameSourceType.PatBinding
+                is RsNamedFieldDecl -> RenameSourceType.Field
+                else -> null
+            }
+
+            return RenameSourceTypeTrackingUsageInfo(
+                referenceElement,
+                ref,
+                ref.rangeInElement.startOffset,
+                ref.rangeInElement.endOffset,
```

Comment:
Why do you need these params?
Looks like it's enough to pass only `ref`

Reply:
I just pass them to `MoveRenameUsageInfo` constructor the same way superclass `RenamePsiElementProcessor` does it. If I don't provide offsets, they will be set to -1, I don't know whether it is ok

Reply:
Ah, I understand, I'll fix